src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1307,57 +1279,54 @@ Scheme Terminal::GetColorScheme() const noexcept
 {
     Scheme s;
 
-    s.Foreground = til::color{ _colorTable.at(TextColor::DEFAULT_FOREGROUND) };
-    s.Background = til::color{ _colorTable.at(TextColor::DEFAULT_BACKGROUND) };
+    s.Foreground = til::color{ _renderSettings.GetColorAlias(ColorAlias::DefaultForeground) };
+    s.Background = til::color{ _renderSettings.GetColorAlias(ColorAlias::DefaultBackground) };
 
     // SelectionBackground is stored in the ControlAppearance
-    s.CursorColor = til::color{ _colorTable.at(TextColor::CURSOR_COLOR) };
-
-    s.Black = til::color{ _colorTable[0] };
-    s.Red = til::color{ _colorTable[1] };
-    s.Green = til::color{ _colorTable[2] };
-    s.Yellow = til::color{ _colorTable[3] };
-    s.Blue = til::color{ _colorTable[4] };
-    s.Purple = til::color{ _colorTable[5] };
-    s.Cyan = til::color{ _colorTable[6] };
-    s.White = til::color{ _colorTable[7] };
-    s.BrightBlack = til::color{ _colorTable[8] };
-    s.BrightRed = til::color{ _colorTable[9] };
-    s.BrightGreen = til::color{ _colorTable[10] };
-    s.BrightYellow = til::color{ _colorTable[11] };
-    s.BrightBlue = til::color{ _colorTable[12] };
-    s.BrightPurple = til::color{ _colorTable[13] };
-    s.BrightCyan = til::color{ _colorTable[14] };
-    s.BrightWhite = til::color{ _colorTable[15] };
+    s.CursorColor = til::color{ _renderSettings.GetColorTableEntry(TextColor::CURSOR_COLOR) };
+
+    s.Black = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_BLACK) };
+    s.Red = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_RED) };
+    s.Green = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_GREEN) };
+    s.Yellow = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_YELLOW) };
+    s.Blue = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_BLUE) };
+    s.Purple = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_MAGENTA) };
+    s.Cyan = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_CYAN) };
+    s.White = til::color{ _renderSettings.GetColorTableEntry(TextColor::DARK_WHITE) };
+    s.BrightBlack = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_BLACK) };
+    s.BrightRed = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_RED) };
+    s.BrightGreen = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_GREEN) };
+    s.BrightYellow = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_YELLOW) };
+    s.BrightBlue = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_BLUE) };
+    s.BrightPurple = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_MAGENTA) };
+    s.BrightCyan = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_CYAN) };
+    s.BrightWhite = til::color{ _renderSettings.GetColorTableEntry(TextColor::BRIGHT_WHITE) };

```

Comment:
FYI Personally I'd do:
```cpp
auto& baseColors = *reinterpret_cast<std::array<Color, 16>*>(&s.Black);
for (size_t i = 0; i < 16; ++i)
{
    baseColors[i] = til::color{ _renderSettings.GetColorTableEntry(i) };
}
```

and
```cpp
const auto& baseColors = *reinterpret_cast<const std::array<Color, 16>*>(&colorScheme.Black);
for (size_t i = 0; i < 16; ++i)
{
    _renderSettings.SetColorTableEntry(i, til::color{ baseColors[i] });
}
```

I mean... It's not like it's our fault that MIDL3 can't do arrays. It practically forced our hand here! We had to reinterpret cast it in order to survive another day in this world! Also code doesn't break code, humans break code. So if we're just careful to not break the code... this reinterpret cast works forever. /s
(In all seriousness though, I'd personally use reinterpret cast here, because it makes the code vastly shorter.)

Reply:
I'm not entirely opposed to this, but if we want to go this route and I think it should be a follow-up PR. There are a bunch of cases like this in the Terminal code, so it's best they're all done at the same time, and I'd rather avoid making too many unrelated changes in this PR.